clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1690,10 +1711,17 @@ void SYCLIntegrationHeader::emitFwdDecl(raw_ostream &O, const Decl *D,
   PrintingPolicy P(D->getASTContext().getLangOpts());
   P.adjustForCPlusPlusFwdDecl();
   P.SuppressTypedefs = true;
+  P.SuppressUnwrittenScope = true;
   std::string S;
   llvm::raw_string_ostream SO(S);
   D->print(SO, P);
-  O << SO.str() << ";\n";
+  O << SO.str();
+
+  if (const auto *ED = dyn_cast<EnumDecl>(D)) {
+    QualType T = ED->getIntegerType();
```

Comment:
Note: 
/// This returns a null QualType for an enum forward definition with no fixed
   /// underlying type.

You likely want to use its promotion type as backup, since that is guaranteed to be able to hold it.

Reply:
Missed this. Sorry

Reply:
Done